src/core/appinterface.cpp

```diff
@@ -353,26 +360,41 @@ void AppInterface::importUrl( const QString &url )
 
             if ( QgsZipUtils::unzip( filePath, zipDirectory, zipFiles, false ) )
             {
-              file.remove();
+              temporaryFile->remove();
+              // Archived project successfully imported
               emit importEnded( zipDirectory );
-              return;
             }
             else
             {
-              // Broken archive, bail out
+              // Broken archive, bail out of importing project
               QDir dir( zipDirectory );
               dir.removeRecursively();
-              file.remove();
+              temporaryFile->remove();
               filePath.clear();
+              emit importEnded();
             }
           }
+          else
+          {
+            // Archived dataset successfully imported
+            emit importEnded( QFileInfo( filePath ).absolutePath() );
+          }
+        }
+        else
+        {
+          // Dataset successfully imported
+          emit importEnded( QFileInfo( filePath ).absolutePath() );
         }
 
-        emit importEnded( QFileInfo( filePath ).absolutePath() );
+        temporaryFile->deleteLater();
+        delete temporaryFile;
         return;
       }
     }
 
+    temporaryFile->deleteLater();
+    delete temporaryFile;
```

Comment:
Does `deleteLater()` have any effect if it's deleted the line after?
Do we need manual memory management here, or could we just use a `QTemporaryFile temporaryFile;` on the stack instead?